*** motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu.c	Sun Aug 23 20:03:57 2020
--- motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu.c	Sun Aug 23 20:01:06 2020
***************
*** 24,29 ****
--- 24,33 ----
  #include <math.h>
  #include "inv_mpu.h"
  
+ #ifdef ESP_PLATFORM
+ #include "../../../../../../hal/include/mpu_idf_esp32.h"
+ #endif
+ 
  /* The following functions must be defined for this platform:
   * i2c_write(unsigned char slave_addr, unsigned char reg_addr,
   *      unsigned char length, unsigned char const *data)
***************
*** 101,106 ****
--- 105,117 ----
  /* UC3 is a 32-bit processor, so abs and labs are equivalent. */
  #define labs        abs
  #define fabs(x)     (((x)>0)?(x):-(x))
+ 
+ #elif defined CONFIG_IDF_TARGET_ESP32
+ #define delay_ms    esp32_delay_ms
+ #define get_ms      esp32_get_clock_ms
+ #define log_i       MPL_LOGI
+ #define log_e       MPL_LOGE
+ 
  #else
  #error  Gyro driver is missing the system layer implementations.
  #endif
*** motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu.h	Sun Aug 23 20:03:57 2020
--- motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu.h	Sun Aug 23 14:54:33 2020
***************
*** 34,39 ****
--- 34,43 ----
      unsigned short pin;
      unsigned char lp_exit;
      unsigned char active_low;
+ #elif defined ESP_PLATFORM
+     unsigned int interrupt_pin;
+     void (*cb)(void*);
+     void * arg;
  #elif defined EMPL_TARGET_UC3L0
      unsigned long pin;
      void (*cb)(volatile void*);
*** motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu_dmp_motion_driver.c	Sun Aug 23 20:03:57 2020
--- motion_driver_6.12/msp430/eMD-6.0/core/driver/eMPL/inv_mpu_dmp_motion_driver.c	Sun Aug 23 20:01:14 2020
***************
*** 24,29 ****
--- 24,33 ----
  #include "dmpKey.h"
  #include "dmpmap.h"
  
+ #ifdef ESP_PLATFORM
+ #include "../../../../../../hal/include/mpu_idf_esp32.h"
+ #endif
+ 
  /* The following functions must be defined for this platform:
   * i2c_write(unsigned char slave_addr, unsigned char reg_addr,
   *      unsigned char length, unsigned char const *data)
***************
*** 59,64 ****
--- 63,74 ----
  #include "uc3l0_clock.h"
  /* delay_ms is a function already defined in ASF. */
  #define get_ms  uc3l0_get_clock_ms
+ #define log_i       MPL_LOGI
+ #define log_e       MPL_LOGE
+ 
+ #elif defined CONFIG_IDF_TARGET_ESP32
+ #define delay_ms    esp32_delay_ms
+ #define get_ms      esp32_get_clock_ms
  #define log_i       MPL_LOGI
  #define log_e       MPL_LOGE
  
*** motion_driver_6.12/msp430/eMD-6.0/core/driver/include/mlinclude.h	Sun Aug 23 20:03:57 2020
--- motion_driver_6.12/msp430/eMD-6.0/core/driver/include/mlinclude.h	Sun Aug 23 19:50:41 2020
***************
*** 6,11 ****
--- 6,13 ----
  #ifndef INV_INCLUDE_H__
  #define INV_INCLUDE_H__
  
+ #pragma GCC diagnostic ignored "-Wunused-local-typedefs"
+ 
  #define INVENSENSE_FUNC_START  typedef int invensensePutFunctionCallsHere
  
  #ifdef COVERAGE
*** motion_driver_6.12/msp430/eMD-6.0/core/driver/include/mltypes.h	Sun Aug 23 20:03:57 2020
--- motion_driver_6.12/msp430/eMD-6.0/core/driver/include/mltypes.h	Sun Aug 23 19:39:26 2020
***************
*** 129,135 ****
  #ifndef __cplusplus
  #ifndef __KERNEL__
  #ifndef EMPL_TARGET_UC3L0
! typedef int_fast8_t bool;
  #endif
  
  #ifndef false
--- 129,135 ----
  #ifndef __cplusplus
  #ifndef __KERNEL__
  #ifndef EMPL_TARGET_UC3L0
! //typedef int_fast8_t bool;
  #endif
  
  #ifndef false
